src/platform/Infineon/crypto/trustm/CHIPCryptoPALHsm_P256_trustm.cpp

```diff
@@ -0,0 +1,601 @@
+{
+#if !ENABLE_TRUSTM_GENERATE_EC_KEY
+    return ECDH_derive_secret_H(&mKeypair, remote_public_key, out_secret);
+#else
+    CHIP_ERROR error = CHIP_ERROR_INTERNAL;
+    optiga_lib_status_t return_status = OPTIGA_LIB_BUSY;
+    size_t secret_length = (out_secret.Length() == 0) ? out_secret.Capacity() : out_secret.Length();
+    uint32_t keyid = 0;
+    if (CHIP_NO_ERROR != get_trustm_keyid_from_keypair(mKeypair, &keyid))
+    {
+        ChipLogDetail(Crypto, "ECDH_derive_secret : Using Host for ECDH");
+        return ECDH_derive_secret_H(&mKeypair, remote_public_key, out_secret);
+    }
+
+    ChipLogDetail(Crypto, "ECDH_derive_secret: Using TrustM for ECDH !");
+    trustm_Open();
+
+    const uint8_t * const rem_pubKey = Uint8::to_const_uchar(remote_public_key);
+    const size_t rem_pubKeyLen       = remote_public_key.Length();
+    
+    uint8_t remote_key[68];
+    uint8_t header[3] = {0x03, 0x42, 0x00};
+
+    memcpy(remote_key, &header, 3);
+    memcpy(remote_key+3, rem_pubKey, rem_pubKeyLen);
+
+    return_status = trustm_ecdh_derive_secret(OPTIGA_KEY_ID_E100, (uint8_t*)remote_key, (uint16_t)rem_pubKeyLen+3, 
+                       out_secret.Bytes(), (uint8_t)secret_length); 
+    
+    VerifyOrExit(return_status == OPTIGA_LIB_SUCCESS, error = CHIP_ERROR_INTERNAL) ;
+
+    exit:
+        if (error != CHIP_NO_ERROR)
+        {
+            trustm_close();
+        }
+        return out_secret.SetLength(secret_length);
+#endif       
+}
+
+CHIP_ERROR P256PublicKey::ECDSA_validate_hash_signature(const uint8_t * hash, size_t hash_length,
+                                                           const P256ECDSASignature & signature) const
+{
+#if !ENABLE_TRUSTM_ECDSA_VERIFY
+    return ECDSA_validate_hash_signature_H(this, hash, hash_length, signature);
+#else    
+    CHIP_ERROR error                                          = CHIP_ERROR_INTERNAL;
+    optiga_lib_status_t return_status                         = OPTIGA_LIB_BUSY;
+    uint8_t signature_trustm[kMax_ECDSA_Signature_Length_Der] = { 0 };
+    size_t signature_trustm_len                               = sizeof(signature_trustm);
+    MutableByteSpan out_der_sig_span(signature_trustm, signature_trustm_len);
+
+    uint8_t hash_length_u8            = static_cast<uint8_t>(hash_length);
+    uint16_t signature_trustm_len_u16 = static_cast<uint16_t>(signature_trustm_len);
+
+    VerifyOrReturnError(hash != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(hash_length > 0, CHIP_ERROR_INVALID_ARGUMENT);
+    ChipLogDetail(Crypto, "ECDSA_validate_hash_signature: Using TrustM for TrustM verify (hash) !");
+
+    // Trust M init
+    trustm_Open();
+    error = EcdsaRawSignatureToAsn1(kP256_FE_Length, ByteSpan{ Uint8::to_const_uchar(signature.ConstBytes()), signature.Length() },
+                                    out_der_sig_span);
+    SuccessOrExit(error);
+
+    /* Set the public key */
+    // P256PublicKeyHSM & public_key = const_cast<P256PublicKeyHSM &>(Pubkey());
```

Comment:
consider deleting commented-out code or have documentation explaining why this code exists as commented-out (is it an example, todo, something else?)

Reply:
Removed the unnecessary  commented-out code.